### ISSUE TITLE
Convert buildSrc/build.gradle to Kotlin

### ISSUE THREAD

**Issue by Bukama** [junit-pioneer/junit-pioneer#282]
In #262 we provided a [buildSrc/build.gradle script](https://github.com/junit-pioneer/junit-pioneer/blob/master/buildSrc/build.gradle) to add IDE specific support for IntelliJ. 

This should be converted to Kotlin as the build.gradle.kts script in root folder.

**Comment by aepfli** [comment]
are we sure about that? i know that we use kotlin in the other scripts - but groovy has its peaks, and in this case, as we are manipulating XML it is way easier to do that with groovy and the DSL for that. Yes we can exclude it into own tasks, but i am not sure it is really worth the effort. maybe the linked tasks of #262 will be soon resolved or there will be a different way. But implementation wise, it will (imho) never be as clean as it is with groovy.

The only thing i could think of, is extracting this into own tasks written in groovy, but i am not sure about this :)

**Comment by Bukama** [comment]
The topic came up at @nicolaiparlog _Midsummer Hackathon_ and the idea was to align the Gradle scripts to one language. As I don't know much about Gradle: Gradle supports both: Kotlin and Groovy, right?

**Comment by aepfli** [comment]
yes, but gradle and kotlin are quiet different in their nature and DSL support. I did not find a nice way to manipulate XML with kotlin

kotlin
```kotlin
    val xmlFile = File("file.xml")
 
    val dbFactory = DocumentBuilderFactory.newInstance()
    val dBuilder = dbFactory.newDocumentBuilder()
    val xmlInput = InputSource(StringReader(xmlFile.readText()))
    val doc = dBuilder.parse(xmlInput)

```

groovy
```
def xmlFile = new File("file.xml")
def xml = new XmlSlurper().parse(xmlFile)
```

and that is just reading - when manipulating it, it gets more and more complicated in kotlin. in groovy i can just access the element with their tag name like `xml.element ` and i even can easier maniuplate it, and/or search through all of them.

It is just way easier to manipulate the xml  in groovy as in kotlin. But as said, maybe we want to just extract the script than into a groovy script and apply it in the build.gradle

**Comment by nipafx** [comment]
I wanna give this a try. 😁

**Comment by keturn** [comment]
Hmm, yeah, if you only have a single function in there and it's using Groovy's XmlSlurper, there's not really a pressing argument to convert it to Kotlin.

Ideally this would be supported by gradle-idea-ext and then you wouldn't have to have this code that makes assumptions about some other application's internal storage format. Sounds like you've already looked in to that: https://github.com/JetBrains/gradle-idea-ext-plugin/issues/88
